code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+
+/// Finds a suitable turf near the given location
+/obj/machinery/big_manipulator/proc/find_suitable_turf_near(turf/center)
+	if(!center)
+		return null
+
+	var/turf/manipulator_turf = get_turf(src)
+	if(!manipulator_turf)
+		return null
+
+	if(center.z != manipulator_turf.z)
+		center = locate(center.x, center.y, manipulator_turf.z)
+		if(!center)
+			return null
+
+	var/list/directions = list(NORTH, EAST, SOUTH, WEST, NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST)
+	for(var/dir in directions)
+		var/turf/check = get_step(center, dir)
+		if(check && !isclosedturf(check))
+			return check
+
+	return null
+
+/// Removes an invalid interaction point from the lists.
+/obj/machinery/big_manipulator/proc/remove_invalid_point(datum/interaction_point/point)
+	if(!point)
+		return
+
+	if(point in pickup_points)
+		pickup_points -= point
+	else if(point in dropoff_points)
+		dropoff_points -= point
+
+	qdel(point)
+	if(is_operational)
+		update_hud()
+
+/obj/machinery/big_manipulator/emag_act(mob/user, obj/item/card/emag/emag_card)
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return FALSE
+
+	balloon_alert(user, "overloaded")
+	obj_flags |= EMAGGED
+
+	update_all_points_on_emag_act()
+	return TRUE
+
+/obj/machinery/big_manipulator/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	default_unfasten_wrench(user, tool, time = 1 SECONDS)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/big_manipulator/can_be_unfasten_wrench(mob/user, silent)
+	if(current_task != CURRENT_TASK_NONE || on)
+		to_chat(user, span_warning("[src] is activated!"))
+		return FAILED_UNFASTEN
+	return ..()
+
+/obj/machinery/big_manipulator/default_unfasten_wrench(mob/user, obj/item/wrench, time)
+	. = ..()
+	if(. == SUCCESSFUL_UNFASTEN)
+		if(anchored) // on anchoring, validate all points and remove invalid ones
+			validate_all_points()
+			update_hud()
+		else
+			remove_all_huds()
+
+/obj/machinery/big_manipulator/screwdriver_act(mob/living/user, obj/item/tool)
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, tool))
+		return ITEM_INTERACT_SUCCESS
+	return ITEM_INTERACT_BLOCKING
+
+/obj/machinery/big_manipulator/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+	return ITEM_INTERACT_BLOCKING
+
+/obj/machinery/big_manipulator/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(user.combat_mode)
+		return NONE
+	if(!panel_open || !is_wire_tool(tool))
+		return NONE
+	wires.interact(user)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/big_manipulator/RefreshParts()
+	. = ..()
+	process_upgrades()
+
+/obj/machinery/big_manipulator/mouse_drop_dragged(atom/drop_point, mob/user, src_location, over_location, params)
+	if(isnull(monkey_worker))
+		return
+
+	if(current_task != CURRENT_TASK_NONE)
+		balloon_alert(user, "turn it off first!")
+		return
+
+	var/mob/living/carbon/human/species/monkey/poor_monkey = monkey_worker.resolve()
+	if(isnull(poor_monkey))
+		return
```

Comment:
why do one of these after the other? both would look the same to a user right? you could make it `worker?.resolve()` and hit em both at once (remember to set monkey_worker to null if it doesn't have anything in it)